AmplifyPlugins/Predictions/AWSPredictionsPlugin/Service/Predictions/AWSPredictionsService+Transcribe.swift

```diff
@@ -31,6 +31,17 @@ extension AWSPredictionsService: AWSTranscribeStreamingServiceBehavior {
         request.mediaSampleRateHertz = 8_000
 
         transcribeClientDelegate.connectionStatusCallback = { status, error in
+            guard status != .closed && error == nil else {
+                let error = error as NSError?
+                guard error?.domain != "NSURLErrorDomain" else {
+                    let predictionsErrorString = PredictionsErrorHelper.mapPredictionsServiceError(error!)
```

Comment:
```

if urlError = error as? URLError, let urlErrorTuple = PredictionsErrorHelper.parseURLError(error) {
   onEvent(.failed(...
}


func parseURLError(_ error: URLError) -> (String, String)
```

Reply:
`PredictionsErrorHelper.mapPredictionsServiceError` can be something more general like `PredictionsErrorHelper.mapError` and does the chain of error handling

Reply:
Even better, add a condition to `PredictionsError.init` that checks whether the incoming error is a `URLError` and take action appropriately. If the use case is such that supplying a description and recovery suggestion isn't appropriate, then consider a type-specific initializer rather than a "Helper" class with a specialized factory method.